aptos-move/move-examples/shared_account/sources/SharedAccount.move

```diff
@@ -0,0 +1,195 @@
+module SharedAccount::SharedAccount {
+    use Std::Errors;
+    use Std::Signer;
+    use Std::Vector;
+    use AptosFramework::Account;
+    use AptosFramework::Coin;
+
+    // struct State records the address of the share_holder and the numerator and the denominator of a fraction
+    struct State has store {
+        share_holder: address,
+        numerator: u64,
+        denominator: u64,
+    }
+
+    // Resource representing a shared account
+    struct SharedAccount has key {
+        share_record: vector<State>,
+        signer_capability: Account::SignerCapability,
+    }
+
+    const EINVALID_INPUT: u64 = 0;
+    const EACCOUNT_NOT_FOUND: u64 = 1;
+    const EADDRESS_ALREADY_EXISTS: u64 = 2;
+    const EINVALID_NUMERATOR_DENOMINATOR_COMBINATIONS: u64 = 3;
+    const EINVALID_SIGNER: u64 = 4;
+    const EINSUFFICIENT_BALANCE: u64 = 5;
+
+    fun find_greatest_common_divisor(denominator1: u64, denominator2: u64) : u64 {
+        if (denominator1 == 0) {
+            denominator2
+        }
+        else {
```

Comment:
nit: else should go to previous line per standard guideline for most languages.

if {
  // code
} else {
  // code
}